dashboard/lib/user_multi_auth_helper.rb

```diff
@@ -24,6 +24,17 @@ def oauth_tokens_for_provider(provider)
     end
   end
 
+  def uid_for_provider(provider)
+    if migrated?
+      AuthenticationOption.find_by(
+        credential_type: provider,
+        user_id: id
+      )&.authentication_id
```

Comment:
`UserMultiAuthHelper` assumes it's included in `User`, so this could be simplified to
```rb
authentication_options.find_by(credential_type: provider)&.authentication_id
```